sdk/ml/azure-ai-ml/azure/ai/ml/dsl/_group_decorator.py

```diff
@@ -152,17 +152,38 @@ def _create_init_fn(cls, fields):  # pylint: disable=unused-argument
         def _get_data_type_from_annotation(anno: Input):
             if isinstance(anno, GroupInput):
                 return anno._group_class
-
-            return IOConstants.PRIMITIVE_STR_2_TYPE[anno.type]
+            # keep original annotation for Outputs
+            if isinstance(anno, Output):
+                return anno
+            try:
+                # convert to primitive type annotation if possible
+                return IOConstants.PRIMITIVE_STR_2_TYPE[anno.type]
+            except KeyError:
+                # otherwise, keep original annotation
+                return anno
+
+        def _has_default(val):
+            # only if val has defaults or it's primitive inputs
+            if isinstance(val, Input) and val.type in IOConstants.PRIMITIVE_STR_2_TYPE:
+                return True
+            if hasattr(val, "default") and val.default is not None:
+                return True
+            return False
 
         locals = {f"_type_{key}": _get_data_type_from_annotation(val) for key, val in fields.items()}
         # Collect field defaults if val is parameter and is optional
-        defaults = {f"_default_{key}": fields[key].default for key in fields}
+        defaults = {
+            f"_default_{key}": fields[key].default for key, val in fields.items() if _has_default(val)
```

Comment:
We may need to refine the order if not all parameters has default. i.e.
class Group:
  a: int =1
  b: int
will raise python exception when generate the init. I delete this part before as we add default None for all parameters.

Reply:
You may refer to the related part here: _inputs_outputs.utils L160, the function will reorder all parameters.
![image](https://user-images.githubusercontent.com/24237253/203888720-6866266b-f5e5-4f2e-9655-4732d7369e55.png)

Reply:
Yes, actually we have done this, None has been added to all non-default parameter. however, the re-order process mainly for calculate the inherit fields, so we have to keep that.

Reply:
I think we could remove the _has_default check and have a test of the following scenario

Reply:
Added None default to outputs, will raise exception when user assigned default value to output, see `test_group_port_defaults`